app/models/manageiq/providers/redhat/infra_manager/provision_workflow.rb

```diff
@@ -92,4 +92,39 @@ def allowed_storages(options = {})
 
     result.select { |s| s.storage_domain_type == "data" }
   end
+
+  def load_allowed_vlans(hosts, vlans)
+    ems = source_ems
+    if update_vnic_profile_supported?(ems)
+      load_v4_allowed_vlans(ems, vlans)
+    else
+      super(hosts, vlans)
+    end
+  end
+
+  def source_ems
+    src = get_source_and_targets
+    load_ar_obj(src[:ems])
+  end
+
+  def update_vnic_profile_supported?(ems)
+    ems.supports_update_vnic_profile?
+  end
+
+  def load_v4_allowed_vlans(ems, vlans)
+    ems.with_provider_connection(:version => 4) do |connection|
+      profiles = connection.system_service.vnic_profiles_service.list
+
+      uid_ems_cluster = VmOrTemplate.find(get_source_vm.id).ems_cluster.uid_ems
+      cluster_networks = connection.system_service.clusters_service.cluster_service(uid_ems_cluster).networks_service.list
+
+      profiles.each do |p|
```

Comment:
The allowed vlan will contains all of the vnic profiles allowed for a vm in a specific cluster.
However, if the user selects a specific host to be run on, and some of the networks are defined as 'optional networks' on ovirt-engine. we should present only the networks which are currently available on the selected host.
Else, the vm will fail to run on that host.